core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerDecision.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import com.linecorp.armeria.common.Response;
+
+/**
+ * A {@link CircuitBreakerDecision} that determines whether a {@link CircuitBreakerRule} reports a response as
+ * a {@link #success()}, {@link #failure()}, {@link #ignore()}s it.
+ * {@link #next()} skips a {@link CircuitBreakerRule} and will lookup next {@link CircuitBreakerRule}s.
+ */
+public final class CircuitBreakerDecision {
```

Comment:
Yeah, I intended to provide an API design similar to `RetryDecision`.
However, enum is a good choice too.

Reply:
I think `RetryDecision` is not enum because of this:
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/client/retry/RetryDecision.java#L35
Which is that a user might create a new instance with a different `Backoff`.
But for `CircuitBreakerDecision`, we don't have such use cases.

Reply:
I thought users might not be interested in this is enum or not.
The important thing is how to use those APIs. Some API they should use `next()` and others are not.
```java
RetryDecision.next() 
// vs
CircuitBreakerDecision.NEXT
// vs
CircuitBreakerDecision.next()
```
By the way, I'm not strong here. Let me migrate this to enum.

Reply:
I'm not strong on this neither. 😆 If you think, the current implementation is better in terms of providing a similar API with `RetryDecision` then it's good as it is.

Reply:
Then let me keep the current implementation.😉